Create gitlab register component template (#174)
janus-idp/software-templates: templates/github/register-github-component/README.md

# Register existing component to Service Catalog

This template is intended to be used as a starting point for registering an existing github repository to the Service Catalog. Both github.com and GitHub Enterprise are supported.

## Prerequisites

> - Ensure the github host is added to the `app-config.yaml` file in the backstage instance

Reviewer: ```suggestion
- Ensure the GitHub host is added to the `app-config.yaml` file in the backstage instance
```

Please update all instances of `GitHub` with the proper name.

- If using a github enterprise instance, make sure the `apiBaseUrl` is set correctly:

  ```yaml
  integrations:
    gitlab:
      - host: github.mycompany.com # this must match the hostname of your gitlab instance
        token: my-github-token # You can use a token or a github app (app will take precedence)
        apps:
          - appId: app-id
            clientId: client-id
            clientSecret: client-secret
            webhookSecret: webhook-secret
            privateKey: |
              -----BEGIN RSA PRIVATE KEY-----
              ...Key content...
              -----END RSA PRIVATE KEY-----
        apiBaseUrl: https://github.mycompany.com/api/v3 # This field is optional if host is `github.com`
        # apiBaseUrl: https://api.github.mycompany.com # This is also a potential api base url for github enterprise
  ```

### Required GitHub permissions

The GitHub application or access token needs to have the following permissions:

- **Repository permissions**

>   - **Contents**: **_Read & write_** - To be able to create a new branch and push files new `catalog-info.yaml` file into it.

Reviewer: ```suggestion
  - **Contents**: **_Read & write_** - To create new branches and push files.
```

>   - **Pull requests**: **_Read & write_** - To be able to create pull requests with generated `catalog-info.yaml`.

Reviewer: ```suggestion
  - **Pull requests**: **_Read & write_** - To create pull requests.
```

>   - **Commit statuses**: **_Read-only_** - To be able to clone private repositories.

Reviewer: ```suggestion
  - **Commit statuses**: **_Read-only_** - To clone private repositories.
```

## Usage

![github-location-info-image](./images/github-location-info.png)

> - `GitHub hostname`: The hostname of your github instance

Reviewer: ```suggestion
- `GitHub Hostname`: The hostname of your GitHub instance
```

Please update other instances and the template as well.

  - Default value: `github.com`
  - For Github Free, Pro & Team, use `github.com`
  - For Github Enterprise, use the hostname of your instance. e.g. `github.mycompany.com`
  - **NOTE**: this hostname MUST exist in the github integrations in the backstage instance's `app-config.yaml` with the correct access token or github app
- `Github Organization`: The owner of the repository (user or organization)

> - `Repository name`: The name of the repository where your component is located

Reviewer: ```suggestion
- `Repository Name`: The name of the repository where your component is located
```

Please update other instances and the template as well.
